### PR TITLE
rocwmma: init at 0.8-5.3.3

### PR DESCRIPTION
**Pull request by Madouura**
###### Description of changes
Tracking: #197885

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/rocwmma/default.nix [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  nativeBuildInputs = [
+    cmake
+    rocm-cmake
+    hip
+  ];
+
+  buildInputs = [
+    rocm-runtime
+    rocm-device-libs
+    rocm-comgr
```
Comment (Flakebi) [inline pkgs/development/libraries/rocwmma/default.nix:85]: The build seems to more or less only copy the header files and that works fine without these dependencies.
I guess they are needed for tests only?

**Review by Flakebi — APPROVED** [review]

**Comment by Madouura** [comment]
Moved ``rocwmma`` in ``all-packages.nix`` to avoid conflict with ``rocrand``

File: pkgs/development/libraries/rocwmma/default.nix [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+, gtest ? null
+, rocblas ? null
+, texlive ? null
+, doxygen ? null
+, sphinx ? null
+, python3Packages ? null
+, buildTests ? false
+, buildSamples ? false
+, buildDocs ? false
+, gpuTargets ? null # gpuTargets = [ "gfx908:xnack-" "gfx90a:xnack-" "gfx90a:xnack+" ... ]
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/rocwmma/default.nix:18]: There should not be added new instances where packages are being overwritten to null to disable features. This is an anti pattern including the asserts below.

Comment (Madouura) [reply pkgs/development/libraries/rocwmma/default.nix:18]: I was under the impression it was to not include those packages and make the closure smaller.
If it's not, I'll change it.

Comment (Madouura) [reply pkgs/development/libraries/rocwmma/default.nix:18]: ``gpuTargets`` is however ``null`` for a different reason.
Is there a reason to change it?
Would it be better to change it to something like ``[ ]``?

Comment (SuperSandro2000) [reply pkgs/development/libraries/rocwmma/default.nix:18]: > I was under the impression it was to not include those packages and make the closure smaller.

It is enough to use lib.optional(s) or if else then for that.

> Would it be better to change it to something like `[ ]`?

That depends on how it should be handled. If only the entries in the list matter than having it by default an empty list is better defined. If there is a difference between null and empty list then it maybe shouldn't be changed.

Comment (Madouura) [reply pkgs/development/libraries/rocwmma/default.nix:18]: There is no discernible difference, so it should be fine.

**Pull request merged**